src/components/button.js

```diff
@@ -0,0 +1,40 @@
+import React from "react"
+import styled from "styled-components"
+import propTypes from 'prop-types'
+
+import Link from "./link"
+
+const ButtonLink = styled(Link)`
```

Comment:
Actually coming back to this, this seems to be the way styled components are. Happy with it if it's what's done lol, idk what I'm doing

Reply:
Styled components significantly simplify adding/removing CSS. I tried using it in a global file and even with 6 components was getting confused.

Reply:
I don't know how it works (or which is better) but for comparison, in the dashboard repo, we seem to use this `.modules.css` thing for each component (e.g. https://github.com/monash-human-power/dashboard/blob/master/client/src/components/NavBar.module.css)

Reply:
Yeah its basically CSS components except within the JS itself, instead of a separate `.module.css` file. This is significantly easier to debug as everything is confined to its class. 

If you add a style such as `H1 { font-size: 1000px }` in a .module.css it will pollute the global CSS namespace when imported.

Reply:
Yeah this is a styled component. I've played around with all the different CSS styles and I found that SASS was my go-to because:
- It mimics the CSS styling closely so moving from normal CSS to SCSS is really simple
- I don't like styling inside the JS files because I feel like it can lead to very large `.js` files and I don't really care about what the component looks like unless I want to adjust the styling (my major gripe about Vue as well).

Happy for you to use styled components since this was the decision you made though. How have you found it?

Reply:
Compared to CSS modules it is much easier to deal with as the majority of components do not have much logic. If they had more logic I would use SASS in separate files.

Reply:
yeah thats fair enough. See how you go with it! I've been meaning to try out LESS as well just to play around with more CSS frameworks. (CSS is a real programming language boiz)

Reply:
Sounds good then, styled CSS!